Package.swift

```diff
@@ -259,7 +259,7 @@ let package = Package(
             dependencies: ["Build", "SPMTestSupport"]),
         .testTarget(
             name: "CommandsTests",
-            dependencies: ["swift-build", "swift-package", "swift-test", "swift-run", "Commands", "Workspace", "SPMTestSupport"]),
+            dependencies: ["swift-build", "swift-package", "swift-test", "swift-run", "Commands", "Workspace", "SPMTestSupport", "Build"]),
```

Comment:
@neonichu @abertelrud wdyt?

Reply:
This looks like the same dependency we ran into trouble with before, where running the unit tests for even libSwiftPMDataModel will now need Build and everything it brings with it (llbuild, etc).  That will make it not possible to be able to build any of the tests on iOS, for example.

But I think it's the exact same issue as the need to test for the `-entry-point-function-name` flag, so we should be able to use the same solution here.

We should make also add something to the Package to check that there is no transitive dependency of the tests on Build, to give an earlier warning.

Reply:
CommandsTests already has a transitive dependency on Build via Commands, so I'm not sure this changes much - are these tests being run on iOS today? 

The check the entry point name tests use:
```
#if swift(<5.5)
        try XCTSkipIf(true, "skipping because host compiler doesn't support '-entry-point-function-name'")
#endif
```
won't work for this because the version of Swift which has the new flag doesn't have a version number yet

Reply:
Oh, I think this might be fine then, what was problematic was adding the `Build` dependency in `SPMTestSupport` so it would be needed for any test suite.

Reply:
Oh sorry, I missed that this was `CommandsTests` and not `SPMTestSupport`.  This is fine, since only the libSwiftPMDataModel tests are run on iOS.  Shelling out on iOS is prohibited anyway, so none of the command-related tests can work.